rta/src/main/java/com/gluonhq/richtextarea/model/Table.java

```diff
@@ -194,28 +194,33 @@ public String addColumnAndGetTableText(int caret, Direction direction) {
                 // add separator to beginning of current column, for each row
                 pos = currentCol == 0 && i == 0 ? 0 : positions.get(i * columns + currentCol - 1) - start;
             }
-            newText = newText.substring(0, pos) + TextBuffer.ZERO_WIDTH_TABLE_SEPARATOR + newText.substring(pos);
+            buffer.insert(new TextUnit(TextBuffer.ZERO_WIDTH_TEXT), pos);
         }
-        return newText;
+        removeLineFeed(buffer);
+        return buffer;
     }
 
-    public String removeColumnAndGetText(int caret) {
+    public UnitBuffer removeColumnAndGetText(int caret) {
         int currentCol = getCurrentColumn(caret);
-        String newText = text.endsWith("\n") ? text.substring(0, text.length() - 1) : text;
+        UnitBuffer buffer = new UnitBuffer(text.getUnitList());
         for (int i = rows - 1; i >= 0; i--) {
             // remove text from current column, for each row
             int posStart = currentCol == 0 && i == 0 ? 0 : positions.get(i * columns + currentCol - 1) - start;
             int posEnd = positions.get(i * columns + currentCol) - start + (currentCol == 0 && i == 0 ? 1 : 0);
-            newText = newText.substring(0, posStart) + newText.substring(posEnd);
+            buffer.remove(posStart, posEnd);
         }
-        return newText;
+        removeLineFeed(buffer);
+        return buffer;
     }
 
     public void printTable() {
         if (!LOGGER.isLoggable(Level.FINE)) {
             return;
         }
-        String tableText = text.replaceAll("" + TextBuffer.ZERO_WIDTH_TABLE_SEPARATOR, "|").replaceAll("\n", "]");
+        String tableText = text.getInternalText()
+                .replaceAll("" + TextBuffer.ZERO_WIDTH_TABLE_SEPARATOR, "|")
+                .replaceAll(TextBuffer.EMOJI_ANCHOR_TEXT, "\u24D4")
```

Comment:
what exactly are we representing emoji has? some comment would be nice.

Reply:
This is a print method, just for logging only.

The table is printed out like:
```
[ ... | ... |
... | ... ]
```

When an emoji anchor is found (there is no emoji, just an invisible unicode character), we just replace it with `\u24D4`, which is a visible `circled letter e`

Reply:
Added some javadoc